### ISSUE TITLE
Coq

### ISSUE THREAD

**Issue by wellecks** [EleutherAI/math-lm#26]
The stack doesn't have meaningful data for Coq- similar to #16 , we can use BigQuery to get Coq data.

**Comment by wellecks** [comment]
working on this in https://github.com/EleutherAI/math-lm/tree/bigquery_coq 

The initial version gives ~38M tokens.

Remaining items:

- [ ] the bigquery query doesn't seem to pick up some common repositories (e.g. Coq-Hott or similar)
- [ ] quality analysis, and corresponding adjustments to the filtering
- [ ] python script to reproduce bigquery query and export (the initial version was done on the UI)

**Comment by wellecks** [comment]
Add deduplication - number of documents after filtering:
- 19528 : no deduplication
- 14774 : document-level dedup
- 12898 : 2048 character chunk dedup (59M tokens)

**Comment by wellecks** [comment]
regarding missing common repositories:

The `bigquery-public-data.github_repos` table is missing 20 of the top 25 most-starred Coq repositories:
```
CompCert
stalin-sort
Coq-HoTT
UniMath
category-theory
fiat-crypto
math-comp
coq-tricks
sail-riscv
metacoq
QuickChick
bedrock2
InteractionTrees
hacspec
analysis
fiat
kami
TypeTheory
disel
coq-library-undecidability
```

To find this, I got a list of the most starred repositories following [this](https://stackoverflow.com/questions/59887185/bigquery-for-github-how-to-get-most-starred-repo-with-a-specific-language). It does a bigquery query on the `githubarchive.month.202212` table. Note that this table does not have file contents.

Then I got a list of all repositories in the `bigquery-public-data.github_repos` table that we are using (this table contains file contents):
```sql
SELECT
f.repo_name,
FROM
`bigquery-public-data.github_repos.files` AS f
WHERE
((f.path LIKE '%.v'))
```

**Comment by wellecks** [comment]
@zhangir-azerbayev so we could either manually download these (i.e. the top _k_ most popular repos) or leave them out. Thoughts?

**Comment by wellecks** [comment]
----

regarding quality: good 👍

**Comment by zhangir-azerbayev** [comment]
I'm in favor of downloading the top k most popular repositories for some large value of k. 

But since we're downloading code via the Github API, maybe we could also locate repos via a github search endpoint if one exists?

**Comment by wellecks** [comment]
the github API simplifies things a lot, and should remove the need for bigquery - working on it now

**Comment by wellecks** [comment]
closed per #29
